### PR TITLE
Function to check if IPADDRESS is private. 

### PR DESCRIPTION
**Pull request by mattcalder-meta**
## Description
Adding the `is_private_ip` UDF that checks whether an IP address is within private or reserved range that is not globally reachable. We take our definitions from what IANA considers *not* "globally reachable" in
[https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml](https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml) and
[https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml](https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml).

## Motivation and Context
When working with network data, filtering private IP address space is a common requirement. Without a standard function to handle it, it has led to numerous instances of:
* Incorrect/partial detection because of the scattered nature of Internet RFCs.
* Code duplication and general clutter to handle ~25 IP address block conditions.

## Test Plan
Added unit tests that check that the boundaries of all include private ranges return `true` while non-private ranges return `false`.

`mvn clean install -Dtest=TestIPSqlFunctions -Dmaven.javadoc.skip=true -DskipUI -T1C -fn -pl presto-main`

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [ ] CI passed.

```
== RELEASE NOTES ==

General Changes
* - Add function :func:`is_private_ip` :pr:`23243`
```

### CODE REVIEW

**Review by steveburnett — DISMISSED** [review]
LGTM! (docs)

Pull branch, reviewed local docs build, looks good. Thanks!

**Comment by steveburnett** [comment]
* Please sign the CLA as described in [this comment](https://github.com/prestodb/presto/pull/23243#issuecomment-2234396597). 

* Add the PR number to the release note entry: 

```
== RELEASE NOTES ==

General Changes
* - Add function :func:`is_private_ip` :pr:`23243`
```

**Review by elharo — COMMENTED** [review]
This needs at least an issue and possibly an RFC

**Comment by mattcalder-meta** [comment]
> This needs at least an issue and possibly an RFC

https://github.com/prestodb/rfcs/pull/16

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestIPSqlFunctions.java [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+{
+    @Test
+    public void testIsPrivateTrue()
+    {
+        /*
+         * IPv4
+         */
+
+        // 0.0.0.0/8 RFC1122: "This host on this network"
+        assertFunction(
```
Comment (aaneja) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestIPSqlFunctions.java:32]: Can we refactor this to use a `org. testng. annotation.DataProvider` for the tests cases instead ?

Comment (mattcalder-meta) [reply presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestIPSqlFunctions.java:32]: Ah, the generated test per input is really nice.

**Comment by kaikalur** [comment]
> This needs at least an issue and possibly an RFC

We add useful UDFs all the time. Don't need RFC for that

**Pull request closed without merging**